On Fri, 18 Jul 2003 09:24:03 +0200 (CEST), Rene Rebe <rene.rebe@gmx.net> said:

> Hi, On: Thu, 17 Jul 2003 18:58:17 +0100,
> alan.rock@glaramara.freeserve.co.uk wrote:

>> In function `calculate_deps32': moduleops_core.c:118:
>> `STT_REGISTER' undeclared (first use in this function)
>> 
>> This is because the version of elf.h supplied by dietlibc doesn't
>> define STT_REGISTER
>> 

> No - it compiled in the past - and it must compile with dietlibc
> since we use it in the first stage of the installation for module
> loading.

OK - try this patch against dietlibc then

--- ./include/elf.new.h	2002-02-21 20:45:21.000000000 +0000
+++ ./include/elf.h	2003-07-18 10:54:48.000000000 +0100
@@ -142,6 +142,8 @@
 #define STT_SECTION	3
 #define STT_FILE	4
 
+#define STT_REGISTER	13		/* Global register reserved to app. */
+
 #define ELF32_ST_BIND(x) ((x) >> 4)
 #define ELF32_ST_TYPE(x) (((unsigned int) x) & 0xf)
 


